### PR TITLE
Improve logging for mods

### PR DESCRIPTION
**Pull request by ASpoonPlaysGames**
- Removes the log spam of "changing mod search path"
- Logs mods as they are loaded, alongside their ConVars, Scripts, ConCommands, etc.
- Warns in the console about errors in the `mod.json` instead of silently skipping things

### CODE REVIEW

**Comment by GeckoEidechse** [comment]
Regarding:

> - Removes the log spam of "changing mod search path"

Do we still keep an indicator from where the mod is loaded from? Not saying we need to keep that message but given that with auto-download ( #362  ) we'll have a second location where we are loading mods from it would probably be useful to know still where the mod is loaded from ^^

**Comment by ASpoonPlaysGames** [comment]
> Do we still keep an indicator from where the mod is loaded from?

Unsure, but I can always add it when the mod is loaded

**Comment by ASpoonPlaysGames** [comment]
> Do we still keep an indicator from where the mod is loaded from?

This has now been done as of https://github.com/R2Northstar/NorthstarLauncher/pull/445/commits/41fa93d0620da5fb3a01354510169dfc06765f86

**Comment by ASpoonPlaysGames** [comment]
> 1. This implies the conavrs were registered even though the mod is disabled

Yh i need to reword that a bit, they were registered from a modloading point of view, but not as far as the game is concerned

> 2. Don't use `goto`

`goto` is better to read than excessive amounts of nested conditionals and indentation. Normally I would split the different bits of modloading into different functions, and then simply `return` in the guard clauses, but splitting it up into different functions seemed unnecessary for something that is only referenced once

**Comment by ASpoonPlaysGames** [comment]
@F1F7Y 's PR to this branch moved things over to their own functions

I have now fixed merge conflicts (which meant parsing InitScript) so I would consider this complete now I think

File: NorthstarDLL/core/filesystem/filesystem.cpp [hunk 1]
```diff
@@ -73,8 +73,6 @@ void SetNewModSearchPaths(Mod* mod)
 	{
 		if ((fs::absolute(mod->m_ModDirectory) / MOD_OVERRIDE_DIR).string().compare(sCurrentModPath))
 		{
-			NS::log::fs->info("Changing mod search path from {} to {}", sCurrentModPath, mod->m_ModDirectory.string());
```
Comment (GeckoEidechse) [inline NorthstarDLL/core/filesystem/filesystem.cpp:76]: Unrelated to PR, can someone explain me the difference between `NS::log::fs` and `spdlog::info`?

I know what the later does, just kinda confused why we also had our custom logger... ^^"

Comment (ASpoonPlaysGames) [reply NorthstarDLL/core/filesystem/filesystem.cpp:76]: We have multiple different loggers for spdlog, all under `NS::log`.
`NS::log::fs` is for filesystem logs, and has the prefix `[FILESYSTM]` in logs
`NS::log::rpak` is for rpak filesystem logs, and has the prefix `[RPAK_FSYS]` in logs
etc. etc.

Basically its so we can get better context about where a line of logging came from. It was part of the log colours PR

Comment (GeckoEidechse) [reply NorthstarDLL/core/filesystem/filesystem.cpp:76]: Aaah, makes sense. So, we use `spdlog` as generic log then? Or why the switch? ^^"

Comment (ASpoonPlaysGames) [reply NorthstarDLL/core/filesystem/filesystem.cpp:76]: Yeah basically. `spdlog` I believe just uses the default logger (which is setup to prefix with `[NORTHSTAR]`)
There may be an `NS::log` variant for the same job, but meh

**Review by Alystrasz — APPROVED** [review]
Confirmed compiling and working in testing.
Meaningful logs will help a lot in tickets.

**Comment by ASpoonPlaysGames** [comment]
image
fifty contributed to this PR, so probably not best if they review

**Comment by Alystrasz** [comment]
Good catch!

**Comment by F1F7Y** [comment]
I think i can improve on this a bit more, but think its good to be merged as is for now

**Comment by GeckoEidechse** [comment]
Imma try to update branch to CMake, hopefully PR survives the change ^^

**Review by GeckoEidechse — APPROVED** [review]
Tested and confirmed:

- [x] Removes the log spam of "changing mod search path"
- [sorta] Logs mods as they are loaded, alongside their ConVars, Scripts, ~~ConCommands~~ (didn't have any example, someone gimme one please <3), etc.
- [x] Warns in the console about errors in the mod.json instead of silently skipping things

Honestly, if other contributors/modders want this fine by me. Some stuff is nice, kinda indifferent about the rest.

**Comment by uniboi** [comment]
just merge it istg

**Pull request merged**